watchdog/watchdog.py

```diff
@@ -0,0 +1,34 @@
+import requests
+from typing import List
+from node_monitor.bot_email import EmailBot
+
+class Watchdog:
+    def __init__(
+            self, 
+            email_bot: EmailBot, 
+            recipients: List[str],
+            node_monitor_url: str) -> None:
+        self.email_bot = email_bot
+        self.recipients = recipients
+        self.node_monitor_url = node_monitor_url
+        
+    def send_notification(self) -> None:
+        subject = "🚨 Node Monitor is Down 🚨"
+        body = "Node Monitor is down. Restart required immediately!"
+        self.email_bot.send_emails(self.recipients, subject, body)
+
+    def check_node_monitor_status(self) -> bool:
+        try:
+            response = requests.get(self.node_monitor_url)
+            data = response.json()
+
+            if data['status'] == 'offline':
+                self.send_notification()
+            else:
+                print('Server is online.')
+
+        except Exception as e:
+            self.send_notification()
+            print(f"An error occurred: {e}")
+
+
```

Comment:
This is way complex.

I think this should remain a simple script. Creating a new module is a lot. You've created a new directory, an `__init__.py`, a `main.py`, a `watchdog.py`, and a `test_watchdog,py`. Not to mention some of the code is way more boilerplate than it needs to be.



What if you create a `watchdog.py` in the root directory of the repo? 
Lay it out like this:

```py

# imports for the required node_monitor files/classes
import c...

# define a watchdog class (if you really want a `Watchdog` class)
# we can also get away with just a for loop for this or a function that gets put into scheduler.
# let's stick with a class though, just so we can encapsulate the c.EMAIL_USERNAME, etc.

class Watchdog:
    def __init__(...):
        scheduler = [scheduler ...]


# instantiate the class or whatever you've called the fn
watchdog = Watchdog(...)

# test it with some assert statements. Make sure that node monitor is available, etc, etc
# Perhaps use patch if you need to. But this is so simple, i think it only needs to be tested on startup.
assert watchdog.node_monitor_is_up() == True
assert watchdog.node_monitor_ip_is_correct() == True.

# Run the thing
if __name__ == "__main__":
    watchdog.start()
    

```
Obviously this is pseudocode and these are terrible tests to implement in this current form. Use your better judgement to improve it.
Thoughts?